### PR TITLE
ENH: Add data.product to schema

### LINKED ISSUE #876: Create a minimal definition for the `product` key in metadata
This key should be what differentiates data as a standard result data product, such that we can filter flexible exports with standardized exports. A minimal product key could look something like this, to be added onto later.

```yml
product:
  type/name: inplace_volumes
  # result_schema: URL to inplace_volumes.json?
```

These fields are subject to further discussion/refinement. This key should only be present when data is exported with a simplified function. It is a "stamp" of data quality.

### PR DESCRIPTION
**Pull request by tnatt**
PR that closes #876

Tests will be added once we add functionality to set the `product` in the metadata

### CODE REVIEW

File: src/fmu/dataio/providers/objectdata/_base.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def __post_init__(self) -> None:
             metadata[usecontent] = getattr(
                 content_model.content_incl_specific, usecontent, None
             )
-
+        metadata["product"] = None
```
Comment (tnatt) [inline src/fmu/dataio/providers/objectdata/_base.py:86]: This is just a placeholder now, functionality to set this will be added in a separate PR

File: src/fmu/dataio/_model/root.py [hunk 2]
```diff
@@ -210,6 +210,7 @@ class FmuResultsJsonSchema(GenerateJsonSchema):
         "data.is_prediction",
         "data.name",
         "data.offset",
+        "data.product",
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_model/root.py:213]: Perhaps be more granular on sub-fields under `data.product` rather than making the whole thing "contractual". That way, some subfields can be assumed stable (indexed in Sumo, used by consumers, etc) while others might still be added/removed/changed.

File: src/fmu/dataio/_model/enums.py [hunk 3]
```diff
@@ -4,6 +4,12 @@
 from typing import Type
 
 
+class ProductType(str, Enum):
+    """The product type of a given data object."""
+
+    inplace_volumes = "inplace_volumes"
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_model/enums.py:10]: This feels more like a product "name" or an "identifier", rather than a "type"?

**Review by mferrera — APPROVED** [review]
🎸
